filelog.go

```diff
@@ -136,28 +163,42 @@ func (w *FileLogWriter) Rotate() {
 	w.rot <- true
 }
 
+// Generate the next filename for rotation
+func (w *FileLogWriter) nextFilename(filename string, suffix string) (string, error) {
+        // Attempt filename.suffix
+	fullName := fmt.Sprintf("%s.%s", filename, suffix)
+	if _, err := os.Stat(fullName); os.IsNotExist(err) {
+		// File does not exist, return it as the next filename
+		return fullName, nil
+	}
+
+	// If necessary, add integer suffix
+	for i := 1; i < 10000; i++ {
+		fullNameWithSuffix := fmt.Sprintf("%s.%s.%04d", filename, suffix, i)
+		if _, err := os.Stat(fullNameWithSuffix); os.IsNotExist(err) {
```

Comment:
I would double-check about `permission denied` being `os.IsNotExist`, that seems really unexpected. I think we should clarify the `next logfile increment` error and include the error from the last call to `Stat`, that would make it clearer to users what's going wrong.